File: domein/src/main/java/be/genealogie/domein/dto/NatuurlijkPersoonDTO.java
package be.genealogie.domein.dto;

import be.genealogie.domein.Geslacht;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class NatuurlijkPersoonDTO implements HeeftNatuurlijkPersoonDto {
    private Long id;
    private String naam;
    private String voornaam;
    private LocalDate geborenOp;
    private String geborenTe;
    private LocalDate overledenOp;
    private String overledenTe;
    private Geslacht geslacht;
    private Integer leeftijd;
    private String sterrenbeeld;

    @Override
    public NatuurlijkPersoonDTO natuurlijkPersoon() {
        return this;
    }
}

File: frontend/src/app/model/natuurlijk-persoon-dto.ts
export class NatuurlijkPersoonDto {
  public constructor(public id: number,
                     public naam: string,
                     public voornaam: string,
                     public geslacht: string,
                     public geborenOp: Date | null,
                     public geborenTe: string | null,
                     public overledenOp: Date | null,
                     public overledenTe: string | null,
                     public leeftijd: number | null,
                     public sterrenbeeld: string | null) {
  }
}
